db/migrate/20250318024953_copy_solved_topic_custom_field_to_discourse_solved_solved_topics.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+#
+class CopySolvedTopicCustomFieldToDiscourseSolvedSolvedTopics < ActiveRecord::Migration[7.2]
+  disable_ddl_transaction!
+
+  BATCH_SIZE = 5000
+
+  def up
+    last_id = 0
+    loop do
+      rows = DB.query(<<~SQL, last_id: last_id, batch_size: BATCH_SIZE)
+        INSERT INTO discourse_solved_solved_topics (
+          topic_id,
+          answer_post_id,
+          topic_timer_id,
+          accepter_user_id,
+          created_at,
+          updated_at
+        )
+        SELECT DISTINCT ON (tc.topic_id)
+          tc.topic_id,
+          CAST(tc.value AS INTEGER),
+          CAST(tc2.value AS INTEGER),
+          COALESCE(ua.acting_user_id, -1),
+          tc.created_at,
+          tc.updated_at
+        FROM topic_custom_fields tc
+        LEFT JOIN topic_custom_fields tc2
+          ON tc2.topic_id = tc.topic_id
+          AND tc2.name = 'solved_auto_close_topic_timer_id'
+        LEFT JOIN user_actions ua
+          ON ua.target_topic_id = tc.topic_id
+          AND ua.action_type = #{UserAction::SOLVED}
```

Comment:
We should hardcode the value of `UserAction::SOLVED` into this migration as this migration is meant for the value of the constant at this point in time only.